### PR TITLE
bulkwriter sample read csv

### PR DESCRIPTION
**Pull request by yhmo**

### CODE REVIEW

File: examples/example_bulkwriter.py [hunk 1]
```diff
@@ -41,65 +44,119 @@
...
-    local_writer.commit()
-    print("test local writer done!")
-    print(local_writer.data_path)
-    return local_writer.data_path
+def test_local_writer_npy(schema: CollectionSchema):
+    print(f"\n===================== test local npy writer ====================")
+    with LocalBulkWriter(
+            schema=schema,
+            local_path="/tmp/bulk_writer",
+            segment_size=4*1024*1024,
```
Comment (yiwen92) [inline examples/example_bulkwriter.py:133]: any reason for hardcode 4 here?

File: examples/example_bulkwriter.py [hunk 2]
```diff
@@ -41,65 +44,119 @@
...
+    ) as local_writer:
+        read_sample_data("./data/train_embeddings.csv", local_writer)
```
Comment (yiwen92) [inline examples/example_bulkwriter.py:135]: Is it possible that csv is too big(100 GB) that cannot load into the processing?

File: examples/example_bulkwriter.py [hunk 3]
```diff
@@ -41,65 +44,119 @@
...
+    local_writer = LocalBulkWriter(
+        schema=schema,
+        local_path="/tmp/bulk_writer",
+        segment_size=1000 * 1024 * 1024,
+        file_type=BulkFileType.JSON_RB,
+    )
     threads = []
     thread_count = 100
-    rows_per_thread = 1000
+    rows_per_thread = 100
```
Comment (yiwen92) [inline examples/example_bulkwriter.py:159]: Any limitation for size per row here?

File: pymilvus/bulk_writer/local_bulk_writer.py [hunk 4]
```diff
@@ -76,20 +95,27 @@ def commit(self, **kwargs):
         x = Thread(target=self._flush, args=(call_back,))
```
Comment (xiaofan-luan) [inline pymilvus/bulk_writer/local_bulk_writer.py:95]: we don't need to flush every append right?
only if buffer is large than expect size? then we change to a new buffer and flush the old buffer async

**Pull request merged**